ludwig/schema/hyperopt/parameter.py

```diff
@@ -0,0 +1,177 @@
+from typing import List, Optional, Type, Union
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.hyperopt.utils import register_parameter_config
+from ludwig.schema.utils import ludwig_dataclass
+
+
+def quantization_number_field(dtype: Union[Type[float], Type[int]] = float, default=None):
+    description = (
+        "Quantization number. Output values will be rounded to the nearest increment of `q` in range."
+        "Quantization makes the upper bound inclusive."
+    )
+    if dtype is int:
+        field = schema_utils.Integer(default=default, allow_none=True, description=description)
+    else:
+        field = schema_utils.Float(default=default, allow_none=True, description=description)
+
+    return field
+
+
+def log_base_field(default=10):
+    return schema_utils.Float(default=default, description="Logarithmic base.")
+
+
+@DeveloperAPI
+@ludwig_dataclass
+class BaseParameterConfig(schema_utils.BaseMarshmallowConfig):
+    space: Optional[str] = None
```

Comment:
Space is required right? Or is it no longer required because the space in each of the subclasses is a predefined ProtectedString?

Reply:
I was on the fence about removing `BaseParameterConfig` and just having the base classes of each parameter variant subclass `schema_utils.BaseMarshmallowConfig`. Since this is a parent to non-overlapping config classes, I think it makes sense to get rid of it.